activestorage/lib/active_storage/attached/many.rb

```diff
@@ -46,10 +46,14 @@ def detach
 
     # Directly purges each associated attachment (i.e. destroys the blobs and
     # attachments and deletes the files on the service).
-    def purge
+    def purge(*args)
       if attached?
-        attachments.each(&:purge)
-        attachments.reload
+        if args.any?
+          attachments.where(id: [args]).each(&:purge)
```

Comment:
`attachments.where(id: args)` (unneeded nested `[args]`) since `args` is an array.

Reply:
Changed in latest commit